internal/service/arcregionswitch/data_source_plan.go

```diff
@@ -0,0 +1,256 @@
+	response.TypeName = request.ProviderTypeName + "_arcregionswitch_plan"
+}
+
+func (d *dataSourcePlan) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = fwschema.Schema{
+		Attributes: map[string]fwschema.Attribute{
+			names.AttrARN: fwschema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					fwvalidators.ARN(),
+				},
+			},
+			names.AttrName: fwschema.StringAttribute{
+				Computed: true,
+			},
+			"execution_role": fwschema.StringAttribute{
+				Computed: true,
+			},
+			"recovery_approach": fwschema.StringAttribute{
+				Computed: true,
+			},
+			"regions": fwschema.ListAttribute{
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			names.AttrDescription: fwschema.StringAttribute{
+				Computed: true,
+			},
+			"primary_region": fwschema.StringAttribute{
+				Computed: true,
+			},
+			"recovery_time_objective_minutes": fwschema.Int64Attribute{
+				Computed: true,
+			},
+			"wait_for_health_checks": fwschema.BoolAttribute{
+				Optional:    true,
+				Description: "Wait for Route53 health check IDs to be populated (takes ~4 minutes)",
+			},
+			"route53_health_checks": fwschema.ListAttribute{
+				ElementType: types.ObjectType{
+					AttrTypes: map[string]attr.Type{
+						"health_check_id":      types.StringType,
+						names.AttrHostedZoneID: types.StringType,
+						"record_name":          types.StringType,
+						names.AttrRegion:       types.StringType,
+					},
+				},
+				Computed:    true,
+				Description: "Route53 health checks associated with the plan",
+			},
+		},
+		Blocks: map[string]fwschema.Block{},
+	}
+}
+
+type dataSourcePlanModel struct {
+	ARN                          types.String `tfsdk:"arn"`
+	Region                       types.String `tfsdk:"region"`
+	Name                         types.String `tfsdk:"name"`
+	ExecutionRole                types.String `tfsdk:"execution_role"`
+	RecoveryApproach             types.String `tfsdk:"recovery_approach"`
+	Regions                      types.List   `tfsdk:"regions"`
+	Description                  types.String `tfsdk:"description"`
+	PrimaryRegion                types.String `tfsdk:"primary_region"`
+	RecoveryTimeObjectiveMinutes types.Int64  `tfsdk:"recovery_time_objective_minutes"`
+	WaitForHealthChecks          types.Bool   `tfsdk:"wait_for_health_checks"`
+	Route53HealthChecks          types.List   `tfsdk:"route53_health_checks"`
+}
+
+func (d *dataSourcePlan) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data dataSourcePlanModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := d.Meta().ARCRegionSwitchClient(ctx)
+
+	plan, err := FindPlanByARN(ctx, conn, data.ARN.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("reading ARC Region Switch Plan", err.Error())
+		return
+	}
+
+	data.ARN = types.StringValue(aws.ToString(plan.Arn))
```

Comment:
For Plugin Framework based resources, AutoFlex can be used rather than individually writing each attribute to state.

https://hashicorp.github.io/terraform-provider-aws/data-handling-and-conversion/#autoflex-for-terraform-plugin-framework-preferred